src/App.tsx

```diff
@@ -17,18 +81,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onFilterChange={handleFilterChange}
+                onSearchChange={handleSearchChange}
+                statusTodo={statusTodo}
+                searchTerm={searchTerm}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loader ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={filteredTodos}
+                  onSelectTodo={handleSelectTodo}
+                  selectedTodo={selectedTodo}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {isModalOpen && selectedTodo && (
```

Comment:
```suggestion
      {selectedTodo && (
```
isModalOpen state is redundant, you can just remove it